### ISSUE TITLE
640x360 resolution fails on Ubuntu

### ISSUE THREAD

**Issue by sergiks** [collab-project/videojs-record#508]
## Description
Requested 640x360 resolution, result is 640x480

## Steps to reproduce
Logitech c930e webcam, desktop with Ubuntu, web page with JS:

    const options = {
        controls: false,
        width: 1280,
        height: 720,
        bigPlayButton: false,
        controlBar: {
            volumePanel: false
        },
        plugins: {
            record: {
                autoMuteDevice: true, 
                audio: false,
                video: {
                    // video media constraints: set resolution of camera
                    minWidth: 640,
                    minHeight: 360,
                    videoMimeType: 'video/mp4',
                },
                // dimensions of captured video frames
                frameWidth: 640,
                frameHeight: 360,

                maxLength: 10,
                displayMilliseconds: false,
                debug: true
            }
        }
    };

    const player = videojs('myVideo', options, function() {
        // print version information at startup
        var msg = 'Using video.js ' + videojs.VERSION +
            ' with videojs-record ' + videojs.getPluginVersion('record') +
            ' and recordrtc ' + RecordRTC.version;
        videojs.log(msg);

        player.record().getDevice();
    });

    player.on('deviceReady', () => {
        // Auto start recording
        player.record().start();
    });

### Expected
Expect to get a video clip in 640x360 resolution.

### Actual
While on Mac OS X result is as expected, on Ubuntu I get video in 640x480 resolution.

### Error output
No errors.

#### videojs
Using video.js 7.8.4 with videojs-record 4.1.0 and recordrtc 5.6.1
#### browsers
Mozilla Firefox, Google Chrome
#### OSes
Ubuntu

Same issue as #408 arised on Ubuntu in Firefox and Google Chrome.

Under Mac OS X and Windows / Firefox and Chrome same webcam Logitech c930e gives requested 640x360 resolution.
But on Ubuntu captured video is 640x480 with the same webcam and code.

`lsusb` and `guvcview` on Ubuntu list the 640x360 among supported resolutions for the webcam, so it is not the driver or hardware issue. [This online JS test](https://webrtchacks.github.io/WebRTC-Camera-Resolution/) also lists the 640x360 resolution as supported ("pass") on Ubuntu. So the issue narrows down to videojs-record, video.js, recordrtc.

Tried suggested code with no success. Both

    video: {
        // video constraints: set resolution of camera
        minWidth: 640,
        minHeight: 360
    },

and

    video: {
        width: 640,
        height: 360
    },

But the resulting video is still 640x480 (not x360) on Ubuntu.

Please advice how to get 640x360 resolution from webcam that supports it.

**Comment by thijstriemstra** [comment]
Thanks for the details, helps a lot.

> Please advice how to get 640x360 resolution from webcam that supports it.

Would be great if I had a camera with that resolution, not sure how to test right now.

I have an old logitech somewhere, I'll check it with https://webrtchacks.github.io/WebRTC-Camera-Resolution/ and hopefully it also supports that resolution.

**Comment by thijstriemstra** [comment]
Looking at the specs: https://www.logitech.com/en-us/product/c930e-webcam

From the specs I could say oh it's not supported on Ubuntu and close this ticket but I obviously won't, getting to support this would be nice.

> This online JS test also lists the 640x360 resolution as supported ("pass") on Ubuntu. So the issue narrows down to videojs-record, video.js, recordrtc.

Can you make a screenshot/table of the results of that page for every OS and various browsers?

**Comment by sergiks** [comment]
I was able to make it work as expected with this syntax:

    video: {
        width: { min: 640, ideal: 640, max: 1280 },
        height: { min: 360, ideal: 360, max: 920 },
    },

I rest my case. Hope this will help someone.

> Can you make a screenshot/table of the results of that page for every OS and various browsers?

Here I created a [table](https://gist.github.com/sergiks/4ac62efa9da3e63e556901b2cc9ed8cc) for FireFox 80 under Mac OS X. Maybe lets limit the range? This one is quite big: browser started freezing generating latter rows. It might have created even more samples if I would wait, perhaps.
